package/samplers/multi_armed_bandit/multi_armed_bandit.py

```diff
@@ -0,0 +1,61 @@
+from collections import defaultdict
+from typing import Any
+from typing import Optional
+
+from optuna.distributions import BaseDistribution
+from optuna.samplers import RandomSampler
+from optuna.study import Study
+from optuna.study._study_direction import StudyDirection
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+class MultiArmedBanditSampler(RandomSampler):
+    """Sampler based on Multi-armed Bandit Algorithm.
+
+    Args:
+        epsilon (float):
+            Params for epsolon-greedy algorithm.
+            epsilon is probability of selecting arm randomly.
+        seed (int):
+            Seed for random number generator and arm selection.
+
+    """
+
+    def __init__(
+        self,
+        epsilon: float = 0.7,
+        seed: Optional[int] = None,
+    ) -> None:
+        super().__init__(seed)
+        self._epsilon = epsilon
+
+    def sample_independent(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        param_name: str,
+        param_distribution: BaseDistribution,
+    ) -> Any:
+        if self._rng.rng.rand() < self._epsilon:
+            return self._rng.rng.choice(param_distribution.choices)
+        else:
+            states = (TrialState.COMPLETE, TrialState.PRUNED)
+            trials = study._get_trials(deepcopy=False, states=states, use_cache=True)
+
+            rewards_by_choice: defaultdict = defaultdict(float)
```

Comment:
[QUESTION] This `defaultdict` treats never choiced arm having 0 rewards. Should i replace any other idea?
(as far as I can think of using `_n_startup_trials` like [TPESampler](https://github.com/ryota717/optuna/blob/326ac8d6402bafdb3d928a99e6ce799aeaed559a/optuna/samplers/_tpe/sampler.py#L407-L410) or letting user set default reward instead of 0)

Reply:
That is a very good point actually:)
I should have given the pseudocode the $\epsilon$-greedy algorithm, but it usually works as follows:
1. The control parameter of the algorithm is $\epsilon$, i.e. the probability of random sampling,  `n_trials`, which we define as $T$ hereafter, and the number of choices $K$.
2. Try every single arm $\epsilon T / K$ times.
3. Choose the optimal arm (up to $\epsilon T / K$ or up to the latest trial) for each dimension.

So usually, we start from the random initialization.
However, we do not have to strictly stick to this algorithm, meaning that it is totally acceptable to not follow the classic algorithm implementation.
Instead, we can do it in the UCB policy fashion where we try each arm once at the initialization.
In this way, your issue will be resolved and we can still retain most of your implementation.

Reply:
Thanks for your suggestion! 

> Instead, we can do it in the UCB policy fashion where we try each arm once at the initialization.

This looks me to good and changed initialization in 371556f
(random initialization seems difficult for Optuna because of its high objective flexibility :pray:)